crates/bevy_animation/src/lib.rs

```diff
@@ -229,24 +256,64 @@ impl AnimationPlayer {
         handle: Handle<AnimationClip>,
         transition_duration: Duration,
     ) -> &mut Self {
-        if self.animation.animation_clip != handle || self.is_paused() {
+        if !self.is_playing_clip(&handle) || self.is_paused() {
             self.start_with_transition(handle, transition_duration);
         }
         self
     }
 
+    /// Handle to the animation clip being played.
+    pub fn animation_clip(&self) -> Option<&Handle<AnimationClip>> {
+        self.animation.animation_clip.as_ref()
+    }
+
+    /// Predicate to check if the given animation clip is being played.
+    pub fn is_playing_clip(&self, handle: &Handle<AnimationClip>) -> bool {
+        if let Some(animation_clip_handle) = self.animation_clip() {
+            animation_clip_handle == handle
+        } else {
+            false
+        }
+    }
+
+    /// Predicate to check if the playing animation has finished, according to the repetition behavior.
+    pub fn is_finished(&self) -> bool {
+        self.animation.finished()
```

Comment:
Should we make both functions named `finished` or `is_finished`?

Reply:
Updated to both use `is_finished()`.